stacks/environment-aws/main.tf

```diff
@@ -51,3 +51,17 @@ provider "vault" {
     }
   }
 }
+
+data "kubernetes_secret" "vault_root_token_secret" {
+  metadata {
+    namespace = module.vault.vault_namespace
+    name      = module.vault.vault_root_token_secret
+  }
+}
+
+provider "vault" {
```

Comment:
I don't have a great suggestion for you, but this is going to cause dependency issues when standing up the environment.  the provider initialization will fail on the first run since the vault instance its pointing to doesn't exist yet.

I'm wondering if now is the time to introduce a layered approach to the code in lead-terraform similar to how we handle the shared service cluster.

Reply:
Yikes, I can see now how this is a problem...will bring it up in standup to get more feedback and thoughts.